### PR TITLE
Added tests, updated deps/code, and fixed compile errors

### PR DESCRIPTION
**Pull request by TheBearodactyl**
Nothing more, nothing less. I tried my best to keep the PR as small as possible this time, while also keeping the commit messages clean.

### CODE REVIEW

File: src/project.rs
Comment (TheBearodactyl) [inline src/project.rs:1]: This was the compile error i got btw.

File: src/util/cache.rs
Comment (TheBearodactyl) [inline src/util/cache.rs:1]: And this was the code I fixed

**Comment by matcool** [comment]
stale, tests are pointless, and no one had that compile error

**Comment by TheBearodactyl** [comment]
> stale, tests are pointless, and no one had that compile error

#1 - that's objectively false, but ok then
#2 - meant to say warning, sorry

**Pull request closed without merging**